frontends/web/src/components/dialog/dialog.tsx

```diff
@@ -115,27 +136,28 @@ class Dialog extends Component<Props, State> {
     }
   };
 
-  private deactivate = () => {
+  public deactivate = () => {
     if (!this.modal.current || !this.overlay.current) {
       return;
     }
-    this.modal.current.classList.remove(style.activeModal);
+    const { isMobile } = this.state;
+    this.modal.current?.classList.remove(style.open);
     this.overlay.current.classList.remove(style.activeOverlay);
-    this.setState({ active: false, currentTab: 0 }, () => {
-      document.removeEventListener('keydown', this.handleKeyDown);
-      if (this.props.onClose) {
-        this.props.onClose();
-      }
-    });
+
+    setTimeout(this.deactivateModal, isMobile ? 350 : 0);
```

Comment:
Hm, the reason we need isMobile here is so that it only delays on mobile, and not for both mobile and desktop. I mean 350ms would probably be somewhat unnoticable delay for most, but it's still a delay that's not needed on desktop (due to no animation) 🙂